exporter/api/msg.go

```diff
@@ -18,91 +22,74 @@ type Message struct {
 }
 
 type SignedMessageAPI struct {
-	Signature types.Signature
-	Signers   []types.OperatorID
+	Signature spectypes.Signature
+	Signers   []spectypes.OperatorID
 	Message   specqbft.Message
 
-	FullData *types.ConsensusData
+	FullData *spectypes.ConsensusData
 }
 
-// NewDecidedAPIMsg creates a new message from the given message
-// TODO: avoid converting to v0 once explorer is upgraded
-func NewDecidedAPIMsg(msgs ...*types.SignedSSVMessage) Message {
-	data, err := DecidedAPIData(msgs...)
-	if err != nil {
-		return Message{
-			Type: TypeDecided,
-			Data: []string{},
-		}
-	}
+type ParticipantsAPI struct {
+	Signers     []spectypes.OperatorID
+	Slot        phase0.Slot
+	Identifier  []byte
+	ValidatorPK string
+	Role        string
+	Message     specqbft.Message
+	FullData    *spectypes.ConsensusData
+}
 
-	decMsg, err := specqbft.DecodeMessage(msgs[0].SSVMessage.Data)
-	if err != nil {
-		return Message{
-			Type: TypeDecided,
-			Data: []string{},
-		}
-	}
-	decMsg2, err := specqbft.DecodeMessage(msgs[len(msgs)-1].SSVMessage.Data)
+// NewParticipantsAPIMsg creates a new message in a new format from the given message.
+func NewParticipantsAPIMsg(msgs ...qbftstorage.ParticipantsRangeEntry) Message {
+	data, err := ParticipantsAPIData(msgs...)
 	if err != nil {
 		return Message{
-			Type: TypeDecided,
+			Type: TypeParticipants,
 			Data: []string{},
 		}
 	}
+	identifier := specqbft.ControllerIdToMessageID(msgs[0].Identifier[:])
+	pkv := identifier.GetDutyExecutorID()
 
-	identifier := specqbft.ControllerIdToMessageID(decMsg.Identifier)
-	// #TODO fixme. its not always pk, it can be a CommitteeID if CommitteeRole or Validator PubKey
-	dutyExecutorID := identifier.GetDutyExecutorID()
-	role := identifier.GetRoleType()
 	return Message{
 		Type: TypeDecided,
 		Filter: MessageFilter{
-			PublicKey: hex.EncodeToString(dutyExecutorID),
-			From:      uint64(decMsg.Height),
-			To:        uint64(decMsg2.Height),
-			Role:      role.String(),
+			PublicKey: hex.EncodeToString(pkv),
+			From:      uint64(msgs[0].Slot),
+			To:        uint64(msgs[len(msgs)-1].Slot),
+			Role:      msgs[0].Identifier.GetRoleType().String(),
```

Comment:
are you sure the messages are sorted by the slot? maybe its best to find the from/to slots in a different way ( loop thorugh and save max and min slot seen )